### PR TITLE
[Task][FE] Implement rooms table list

### PR DESCRIPTION
**Pull request by alOletres**
# Issue Link
[Link to the corresponding issue in the github](https://github.com/orgs/Web-Masters-Freelancing/projects/1/views/1?pane=issue&itemId=52099831)

# Command
`npm run dev`

# Definition of Done
- [x] Can view table display

# Notes
None

# Pre-conditions
None

# Expected Output
- [x] Rooms has Table view list

# Screenshots/Recordings

image

### CODE REVIEW

File: web/src/components/Modal.tsx [hunk 1]
```diff
@@ -28,6 +29,8 @@ const saveButtonWrapper: SxProps<Theme> = {
   marginTop: 1,
 };
 
+export type ButtonName = "Save" | "Save changes";
```
Comment (LodiAniez) [inline web/src/components/Modal.tsx:32]: Dili naman ni needed, if magbutang man gani tag inani na type, istatic nalang diay nato?

File: web/src/app/rooms/hooks.ts [hunk 2]
```diff
@@ -4,29 +4,133 @@ import { Field, InputFieldProps } from "@/components/hooks/useModal";
...
   keyword: string;
 };
 
+interface Schema extends RoomsFormValues, TableActions {}
+
 export const useHooks = () => {
   const { handleCreateRoom } = useRoomApi();
   const { setSnackbarProps } = useSnackbar();
-
+  const [btnName, setBtnName] = useState<ButtonName>("Save");
```
Comment (LodiAniez) [inline web/src/app/rooms/hooks.ts:27]:
```suggestion
  const [btnName, setBtnName] = useState("Save");
```

File: web/src/app/rooms/hooks.ts [hunk 3]
```diff
@@ -4,29 +4,133 @@ import { Field, InputFieldProps } from "@/components/hooks/useModal";
...
   const [open, setOpen] = useState(false);
+  const [title, setTitle] = useState("CREATE ROOMS");
 
-  const initialValues: RoomsFormValues = {
+  const [initialValues, setInitialValues] = useState<RoomsFormValues>({
+    id: undefined,
```
Comment (LodiAniez) [inline web/src/app/rooms/hooks.ts:32]: Pwede mani iremove since naka optional mani sa type

File: web/src/app/rooms/hooks.ts [hunk 4]
```diff
@@ -57,7 +161,6 @@ export const useHooks = () => {
 
       toggleModal();
     } catch (e: any) {
-      console.error(e);
```
Comment (LodiAniez) [inline web/src/app/rooms/hooks.ts:60]: Ayaw ni iremove, incase naay error, ma check nato instantly sa browser logs

File: web/src/app/rooms/hooks.ts [hunk 5]
```diff
@@ -68,58 +171,54 @@ export const useHooks = () => {
...
-        type: "number",
-        margin: "dense",
-      },
+      name: "Edit",
+      variant: "contained",
+      handleClick: handleEdit,
     },
   ];
 
+  const handleSubmit = btnName === "Save" ? handleSave : handleSaveChanges;
```
Comment (LodiAniez) [inline web/src/app/rooms/hooks.ts:209]:
```suggestion
  const handleSubmit = useCallback(() => btnName === "Save" ? handleSave : handleSaveChanges, [btnName]);
```

File: web/src/app/rooms/page.tsx [hunk 6]
```diff
@@ -1,86 +1,114 @@
 "use client";
-import { Box, Button, Card, CardHeader } from "@mui/material";
+import {
+  Box,
+  Button,
+  Card,
+  CardHeader,
```
Comment (LodiAniez) [inline web/src/app/rooms/page.tsx:6]: iremove ni kay wala gigamit

File: web/src/app/rooms/hooks.ts [hunk 7]
```diff
@@ -4,29 +4,133 @@ import { Field, InputFieldProps } from "@/components/hooks/useModal";
 import { useRoomApi } from "@/hooks/api/room";
 import { useSnackbar } from "@/hooks/useSnackbar";
 import { OptionSelect, SelectFieldProps } from "@/components/Select";
+import { ActionButtonProps, Column, TableActions } from "@/components/Table";
+import { ButtonName } from "@/components/Modal";
```
Comment (LodiAniez) [inline web/src/app/rooms/hooks.ts:8]: Dili nani needed

**Review by LodiAniez — CHANGES_REQUESTED** [review]

File: web/src/components/Modal.tsx (hunk 1 above)
Comment (alOletres) [reply web/src/components/Modal.tsx:32]: ge2x

**Review by LodiAniez — APPROVED** [review]

**Pull request merged**